spec/controllers/root_controller_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe RootController, type: :controller do
+  describe 'GET #index' do
+    before(:each) { get :index }
+    it 'returns a status code of 200' do
+      expect(response.status).to eq 200
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template('index')
+    end
+
+    it 'has valid instance variables' do
```

Comment:
**Code Quality -Nitpick**
Consider splitting the assertions into their own examples with specific messages. Remember, your tests should be as expressive as possible.

```ruby
it "creates a new link"
it "assigns top users to @top_users"
it "assigns top links to @top_links"
it "assigns recent links to @recent_links"
```

Also assert the instance variable contains specific objects in addition to  not being nil.

Check this out http://betterspecs.org/